### PR TITLE
Ab/cloud azure plugin unit tests

### PR DESCRIPTION
**Pull request by aslanbekirov**

### CODE REVIEW

File: plugins/cloud-azure/src/main/java/org/elasticsearch/cloud/azure/management/AzureComputeService.java [hunk 1]
```diff
@@ -20,9 +20,11 @@
 package org.elasticsearch.cloud.azure.management;
 
 import com.microsoft.azure.management.compute.models.VirtualMachine;
+import com.microsoft.azure.management.network.models.Subnet;
 import com.microsoft.windowsazure.Configuration;
 
 import java.util.ArrayList;
+import java.util.List;
```
Comment (bogensberger) [inline plugins/cloud-azure/src/main/java/org/elasticsearch/cloud/azure/management/AzureComputeService.java]: remove unused imports

File: plugins/cloud-azure/src/main/java/org/elasticsearch/cloud/azure/management/AzureComputeService.java [hunk 2]
```diff
@@ -20,9 +20,11 @@
 package org.elasticsearch.cloud.azure.management;
 
 import com.microsoft.azure.management.compute.models.VirtualMachine;
+import com.microsoft.azure.management.network.models.Subnet;
```
Comment (bogensberger) [inline plugins/cloud-azure/src/main/java/org/elasticsearch/cloud/azure/management/AzureComputeService.java]: remove unused import

File: plugins/cloud-azure/src/main/java/org/elasticsearch/cloud/azure/management/AzureComputeServiceImpl.java [hunk 3]
```diff
@@ -23,24 +23,30 @@
...
 import com.microsoft.windowsazure.Configuration;
+import com.microsoft.windowsazure.exception.ServiceException;
 import com.microsoft.windowsazure.management.configuration.ManagementConfiguration;
 import org.elasticsearch.ElasticsearchException;
 import org.elasticsearch.cloud.azure.AzureServiceDisableException;
 import org.elasticsearch.cloud.azure.AzureServiceRemoteException;
 import org.elasticsearch.common.component.AbstractLifecycleComponent;
 import org.elasticsearch.common.inject.Inject;
 import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.discovery.azure.AzureDiscovery;
```
Comment (bogensberger) [inline plugins/cloud-azure/src/main/java/org/elasticsearch/cloud/azure/management/AzureComputeServiceImpl.java]: remove import

File: plugins/cloud-azure/src/main/java/org/elasticsearch/cloud/azure/management/AzureComputeServiceImpl.java [hunk 4]
```diff
@@ -23,24 +23,30 @@
 import com.microsoft.azure.management.compute.ComputeManagementClient;
 import com.microsoft.azure.management.compute.ComputeManagementService;
 import com.microsoft.azure.management.compute.models.VirtualMachine;
+import com.microsoft.azure.management.network.NetworkResourceProviderClient;
+import com.microsoft.azure.management.network.NetworkResourceProviderService;
+import com.microsoft.azure.management.network.models.Subnet;
```
Comment (bogensberger) [inline plugins/cloud-azure/src/main/java/org/elasticsearch/cloud/azure/management/AzureComputeServiceImpl.java]: remove imports

File: plugins/cloud-azure/src/main/java/org/elasticsearch/cloud/azure/management/AzureComputeServiceImpl.java [hunk 5]
```diff
@@ -64,16 +70,16 @@ public AzureComputeServiceImpl(Settings settings) {
...
-                (URI)null,
-                subscriptionId, // subscription id
-                authRes.getAccessToken()
+                    null,
+                    (URI) null,
+                    subscriptionId, // subscription id
+                    authRes.getAccessToken()
             );
         } catch (Exception e) {
             logger.error("can not start azure client: {}", e.getMessage());
```
Comment (bogensberger) [inline plugins/cloud-azure/src/main/java/org/elasticsearch/cloud/azure/management/AzureComputeServiceImpl.java]: remove reformattings of code

File: plugins/cloud-azure/src/main/java/org/elasticsearch/discovery/azure/AzureUnicastHostsProvider.java [hunk 6]
```diff
@@ -98,7 +100,7 @@ public AzureUnicastHostsProvider(Settings settings, AzureComputeService azureCom
         HostType tmpHostType = HostType.fromString(strHostType);
         if (tmpHostType == null) {
             logger.warn("wrong value for [{}]: [{}]. falling back to [{}]...", Discovery.HOST_TYPE,
-                strHostType, HostType.PRIVATE_IP.name().toLowerCase(Locale.ROOT));
+                    strHostType, HostType.PRIVATE_IP.name().toLowerCase(Locale.ROOT));
```
Comment (bogensberger) [inline plugins/cloud-azure/src/main/java/org/elasticsearch/discovery/azure/AzureUnicastHostsProvider.java]: revert reformatting

File: plugins/cloud-azure/src/main/java/org/elasticsearch/discovery/azure/AzureUnicastHostsProvider.java [hunk 7]
```diff
@@ -136,99 +138,38 @@ public AzureUnicastHostsProvider(Settings settings, AzureComputeService azureCom
...
-                                    version.minimumCompatibilityVersion()));
-                            }
-                        } catch (Exception e) {
-                            logger.warn("can not convert [{}] to transport address. skipping. [{}]", networkAddress, e.getMessage());
-                        }
+            List<String> ipAddresses = listIPAddresses(networkResourceProviderClient, rgName, networkNameOfCurrentHost.get(AzureDiscovery.VNET),
+                    networkNameOfCurrentHost.get(AzureDiscovery.SUBNET), discoveryMethod, hostType, logger);
+            for (String networkAddress : ipAddresses) {
+                try {
+                    // we only limit to 1 port per address, makes no sense to ping 100 ports
```
Comment (bogensberger) [inline plugins/cloud-azure/src/main/java/org/elasticsearch/discovery/azure/AzureUnicastHostsProvider.java]: /s limit to only one port per address

File: plugins/cloud-azure/src/main/java/org/elasticsearch/discovery/azure/AzureUnicastHostsProvider.java [hunk 8]
```diff
@@ -136,99 +138,38 @@ public AzureUnicastHostsProvider(Settings settings, AzureComputeService azureCom
...
+                    TransportAddress[] addresses = transportService.addressesFromString(networkAddress, 1);
+                    for (TransportAddress address : addresses) {
+                        logger.trace("adding {}, transport_address {}", networkAddress, address);
+                        cachedDiscoNodes.add(new DiscoveryNode("#cloud-" + networkAddress, address,
```
Comment (bogensberger) [inline plugins/cloud-azure/src/main/java/org/elasticsearch/discovery/azure/AzureUnicastHostsProvider.java]: the generated nodename is ok as a temporary solution, but please ensure to use the correct host name later

File: plugins/cloud-azure/src/main/java/org/elasticsearch/discovery/azure/AzureUnicastHostsProvider.java [hunk 9]
```diff
@@ -136,99 +138,38 @@ public AzureUnicastHostsProvider(Settings settings, AzureComputeService azureCom
...
+                                version.minimumCompatibilityVersion()));
                     }
-
+                } catch (Exception e) {
+                    logger.warn("can not convert [{}] to transport address. skipping. [{}]", networkAddress, e.getMessage());
```
Comment (bogensberger) [inline plugins/cloud-azure/src/main/java/org/elasticsearch/discovery/azure/AzureUnicastHostsProvider.java]: is it really necessary to catch Exception? Isn't it possible to do it more specific?

File: plugins/cloud-azure/src/main/java/org/elasticsearch/discovery/azure/AzureUnicastHostsProvider.java [hunk 10]
```diff
@@ -136,99 +138,38 @@ public AzureUnicastHostsProvider(Settings settings, AzureComputeService azureCom
...
                 }
             }
         } catch (UnknownHostException e) {
             logger.error("Error occurred in getting hostname");
-        } catch (ServiceException e) {
-            logger.error("Error occurred in getting public address of NIC");
-        } catch (IOException e) {
-            logger.error("Error occurred in IO operations");
         } catch (Exception e) {
-            logger.error("Error occurred in retrieving IP addresses");
```
Comment (bogensberger) [inline plugins/cloud-azure/src/main/java/org/elasticsearch/discovery/azure/AzureUnicastHostsProvider.java]: same here (catch exception? )

File: plugins/cloud-azure/src/main/java/org/elasticsearch/discovery/azure/AzureUnicastHostsProvider.java [hunk 11]
```diff
@@ -265,4 +206,80 @@ public AzureUnicastHostsProvider(Settings settings, AzureComputeService azureCom
...
+
+                                networkAddress = NetworkAddress.formatAddress(InetAddress.getByName(pip.getIpAddress()));
+                            }
+
+                            if (networkAddress == null) {
+                                logger.trace("no public ip provided. ignoring [{}]...", nic.getName());
+                            }
+                            break;
+                        default:
+                            // This could never happen!
```
Comment (bogensberger) [inline plugins/cloud-azure/src/main/java/org/elasticsearch/discovery/azure/AzureUnicastHostsProvider.java]: dead code, remove it

File: plugins/cloud-azure/src/test/java/org/elasticsearch/discovery/azure/AzureUnicastHostsProviderTest.java [hunk 12]
```diff
@@ -0,0 +1,192 @@
...
+        subnet2.setName("mySubnet2");
+
+        NetworkResourceProviderClient providerClient = mock(NetworkResourceProviderClient.class);
+        VirtualNetworkOperations virtualNetworkOperations = mock(VirtualNetworkOperationsImpl.class);
+        VirtualNetworkGetResponse virtualNetworkGetResponse = mock(VirtualNetworkGetResponse.class);
+        NetworkInterfaceOperations networkInterfaceOperations = mock(NetworkInterfaceOperationsImpl.class);
+        NetworkInterfaceGetResponse networkInterfaceGetResponse = mock(NetworkInterfaceGetResponse.class);
+        NetworkInterfaceGetResponse networkInterfaceGetResponse2 = mock(NetworkInterfaceGetResponse.class);
+        PublicIpAddressOperations  publicIpAddressOperations = mock(PublicIpAddressOperationsImpl.class);
+        PublicIpAddressGetResponse publicIpAddressGetResponse = mock(PublicIpAddressGetResponse.class);
```
Comment (bogensberger) [inline plugins/cloud-azure/src/test/java/org/elasticsearch/discovery/azure/AzureUnicastHostsProviderTest.java]: move the mocking and setUp stuff both tests have in common to setUp method so, to shorten the tests

File: plugins/cloud-azure/src/main/java/org/elasticsearch/cloud/azure/management/AzureComputeServiceImpl.java (hunk 5 above)
Comment (aslanbekirov) [inline plugins/cloud-azure/src/main/java/org/elasticsearch/cloud/azure/management/AzureComputeServiceImpl.java]: I think it is default formatting, I mean correct formatting.

File: plugins/cloud-azure/src/main/java/org/elasticsearch/discovery/azure/AzureUnicastHostsProvider.java (hunk 6 above)
Comment (aslanbekirov) [inline plugins/cloud-azure/src/main/java/org/elasticsearch/discovery/azure/AzureUnicastHostsProvider.java]: same here

**Comment by bogensberger** [comment]
ack

**Pull request merged**